addons/website_sale/controllers/main.py

```diff
@@ -200,6 +200,14 @@ def shop(self, page=0, category=None, search='', **post):
 
         product_obj = pool.get('product.template')
 
+        def recursive_category(category, parent_category=None):
+            if parent_category is None:
```

Comment:
while category.parent_id:
  categories |= category.parent_id (to translate into old API)
  category = category.parent_id